common/rpc/encryption/localStoreTlsFactory.go

```diff
@@ -50,36 +52,58 @@ type localStoreTlsProvider struct {
 
 func NewLocalStoreTlsProvider(tlsConfig *config.RootTLS) (TLSConfigProvider, error) {
 	return &localStoreTlsProvider{
-		internodeCertProvider: &localStoreCertProvider{tlsSettings: &tlsConfig.Internode},
-		frontendCertProvider:  &localStoreCertProvider{tlsSettings: &tlsConfig.Frontend},
-		RWMutex:               sync.RWMutex{},
-		settings:              tlsConfig,
+		internodeCertProvider:              &localStoreCertProvider{tlsSettings: &tlsConfig.Internode},
+		frontendCertProvider:               &localStoreCertProvider{tlsSettings: &tlsConfig.Frontend},
+		frontendPerHostCertProviderFactory: newLocalStorePerHostCertProviderFactory(tlsConfig.Frontend.PerHostOverrides),
```

Comment:
As a follow-up, we probably need to add a server option for injecting other implementations of `PerHostCertProviderFactory`. Although I wonder to what extent it would make sense as a standalone component outside of the already existing option for `TLSConfigProvider`.